lib/ain-grpc/src/rpc/eth.rs

```diff
@@ -297,8 +298,13 @@ impl MetachainRPCModule {
             // BlockNumber::Pending => todo!(),
             _ => self.handler.storage.get_latest_block(),
         }
-        .ok_or(Error::Custom(String::from("header not found")))
-        .map(|block| block.header.number)
+    }
+
+    fn try_get_block_number(&self, block_number: Option<BlockNumber>) -> Result<U256, BlockError> {
```

Comment:
Why do we need to wrap this method with the get block implementation which is fallible? We should have a single try_get_block_by_number method that returns BlockError consistently through all RPCs, and propagated accordingly

Reply:
> Why do we need to wrap this method with the get block implementation which is fallible?

hmm... just to have a utils to get block number easily... and to change fallible from `Option<T>` to `Result<T>`

> We should have a single try_get_block_by_number method 

afraid this one requires more refactoring 😅 

> that returns BlockError consistently through all RPCs, and propagated accordingly

the `get_block() -> Option<T>` is actually following ethereum getBlock behavior (throwing null instead of error)
```
➜  metachain-ts-tests git:(main) ✗ curl https://eth.meowrpc.com \
  -H 'content-type:application/json' \
  --data-binary \
  '{
    "jsonrpc":"2.0",
    "id":"2",
    "method":"eth_getBlockByNumber",
    "params":["0x20f3e96",false]
  }'
{"jsonrpc": "2.0", "id": "2", "result": null}%
```